src/library/scala/Function3.scala

```diff
@@ -36,8 +36,8 @@ trait Function3[-T1, -T2, -T3, +R] extends AnyRef { self =>
    *  @return   a function `f` such that `f((x1, x2, x3)) == f(Tuple3(x1, x2, x3)) == apply(x1, x2, x3)`
    */
 
-  @annotation.unspecialized def tupled: Tuple3[T1, T2, T3] => R = {
-    case Tuple3(x1, x2, x3) => apply(x1, x2, x3)
+  @annotation.unspecialized def tupled: ((T1, T2, T3)) => R = {
+    case ((x1, x2, x3)) => apply(x1, x2, x3)
```

Comment:
not sure should these parameters be `t1,t2...`,so does the `apply` method's `v1,v2,v3...`

Reply:
I think the additional `()` is helpful.